weatherAplication/Modules/Search/ViewModel/SearchViewModel.swift

```diff
@@ -29,12 +31,20 @@ class SearchViewModel: SearchViewModelContract {
                 cityInfoArray = []
                 delegate?.reloadCityNames()
         } else {
-            apiClient.searchCities(searchTerm: searchTerm, completion: { [weak self] cityInfo -> Void in
-                self?.cityInfoArray = cityInfo
-                self?.delegate?.reloadCityNames()
-            }, failure: { weatherError in
-                self.delegate?.showAlert(description: weatherError.localizedDescription)
-            })
+            cancellable = apiClient.searchCity(searchTerm: searchTerm)
+                .sink(
+                    receiveCompletion: { completion in
+                        switch completion {
+                        case.failure(let error):
+                            self.delegate?.showAlert(description: error.localizedDescription)
+                        case.finished:
+                            print("finished")
+                        }
+                    }, receiveValue: {[weak self] cityInfo in
+                        self?.cityInfoArray = cityInfo
+                        self?.delegate?.reloadCityNames()
+                    })
+
         }
```

Comment:
tu nie musisz przypisywac do cancellable, jak masz zbior `private var subscriptions = Set<AnyCancellable>()` to za sinkiem dajesz `.store(in: &subscriptions)`